src/OSPSuite.R/Services/ConcurrentSimulationRunner.cs

```diff
@@ -109,7 +109,7 @@ private Task initializeBatches()
                settings => Enumerable.Range(0, settings.MissingBatchesCount).Select(_ => settings)
             ).ToList(),
             data => new Guid().ToString(),
-            (core, ct, settings) => Task.FromResult(settings.AddNewBatch())
```

Comment:
This is not run on a new thread. That's why it was done sequentially

Reply:
Yes, good catch